lambda/rls-lambda/index.js

```diff
@@ -0,0 +1,246 @@
+    );
+  }
+
+  const realmClients = await httpsRequest(
+    "GET",
+    `/auth/admin/realms/${REALM_NAME}/clients`
+  );
+
+  const qsClient = realmClients.filter(
+    (client) => client.clientId === QUICKSIGHT_CLIENT_NAME
+  );
+
+  const awsClient = realmClients.filter(
+    (client) => client.clientId === AWS_CLIENT_NAME
+  );
+
+  const users = await httpsRequest(
+    "GET",
+    `/auth/admin/realms/${REALM_NAME}/users`
+  );
+  var rlsMap = [];
+
+  for (const user of users) {
+    const quickSightRoles = await httpsRequest(
+      "GET",
+      `/auth/admin/realms/${REALM_NAME}/users/${user["id"]}/role-mappings/clients/${qsClient[0].id}/composite`
+    );  // For each user check what role user has access to based on the quicksight client
+
+    if (quickSightRoles.length > 0) {
+      for (const role of quickSightRoles) {
+        const quickSightRoleName = role.name.split(",")[0].split("/")[1];
+        console.log(quickSightRoleName)
+
+        // const awsAccountIds = await httpsRequest(
+        //   "GET",
+        //   `/auth/admin/realms/${REALM_NAME}/users/${user["id"]}/role-mappings/clients/${awsClient[0].id}/composite`
+        // ) // For each user check what aws client roles does the user has access to.
+        //   .then((roles) => {
+        //     return roles.map((role) => role.name.split(":")[4]);
+        //   })
+        //   .then((roles) => {
+        //     return [...new Set(roles)];
+        //   });
+
+        // rlsMap[`${quickSightRoleName}/${user.email}`] = awsAccountIds;
+
+        //modify code from here 
+        // For each user check what AWS client roles does the user has access to.
+          const awsRoles = await httpsRequest(
+            "GET",
+            `/auth/admin/realms/${REALM_NAME}/users/${user["id"]}/role-mappings/clients/${awsClient[0].id}/composite`
+          );
+          console.log(awsRoles)
+          // Filter roles to only include those containing 'WORKLOAD_billing_viewer' in their name.
+          const filteredRoles = awsRoles.filter(role => role.name.includes(BCGOV_ROLES_FOR_ACCESS));
+
+          // Extract account IDs from the filtered roles.
+          const awsAccountIds = filteredRoles.map(role => {
+            const match = role.name.match(/arn:aws:iam::(\d+):role\/[^,]+/);
+            return match ? match[1] : null;
+          }).filter(accountId => accountId !== null); // Ensure null values are removed.
+
+          // Deduplicate account IDs
+          const uniqueAwsAccountIds = [...new Set(awsAccountIds)];
+
+          rlsMap[`${quickSightRoleName}/${user.email}`] = uniqueAwsAccountIds;
+
+      }
+    }
+  }
+
+  let csvData = "";
+  for (const [username, accounts] of Object.entries(rlsMap)) {
+    if (accounts.length > 1) {
+      csvData += `${username},"${accounts.join(",")}"\n`;
+    } else {
+      csvData += `${username},${accounts.join(",")}\n`;
+    }
+  }
+
+  console.log("CSV data:\n", csvData);
+  // try {
```

Comment:
looks like this isn't needed since you are uploading the data directly further down, you can remove this commented out code